### DISCUSSION TITLE
Launch screen and a progress bar

### DISCUSSION THREAD

**Discussion by Seraphel** [skiptools/skip#126] [Q&A] [answered]
Hi, I would like to find out, (as I am new to Skip and Kotlin), how to design the launch screen for the Android Version. 

And also, the simplest way to implement a progress bar for my Android `web view` based off: 

https://github.com/skiptools/skipapp-bookings/blob/main/Sources/TravelBookings/WebView.swift

Any help would be greatly appreciated. 🙏🏻🙂

**Comment by dfabulich** [comment]
I'm also looking for support for a launch screen. On iOS, you’d use a Launch Screen Storyboard and configure it as your UILaunchScreen. On Android, there’s a SplashScreen API instead. How does this work in Skip?

**Comment by marcprux** [comment, marked as answer]
These sort of early lifecycle/app configuration setups are beyond SkipUI, so any programmatic SplashScreen code would need be implemented in Kotlin in the `Android/app/src/main/kotlin/Main.kt`, or if you just want a static launch screen, you can add the necessary resources to the `Android/app/src/main/res/` folders, as per the [SplashScreen](https://developer.android.com/develop/ui/views/launch/splash-screen) documentation. Note that many of the concepts that apply to Flutter's [splash screen](https://docs.flutter.dev/platform-integration/android/splash-screen) documentation would apply to Skip as well.

We would be interested in hearing the results of your investigation, since it would make a nice addition to the documentation.

> And also, the simplest way to implement a progress bar for my Android web view based off:

You might want to consider basing your browser implementation of the more advanced [skip-web](https://github.com/skiptools/skip-web) framework. If don't know if there is enough delegate information in there to handle a progress bar, but it might not be too much work to add one. The Net Skip app uses a progress bar (see https://github.com/Net-Skip/Net-Skip/blob/main/Sources/NetSkip/BrowserView.swift#L167), but I think it currently only works on the iOS side.
